flink-yarn/src/test/java/org/apache/flink/yarn/entrypoint/YarnEntrypointUtilsTest.java

```diff
@@ -0,0 +1,112 @@
+import org.apache.flink.util.TestLogger;
+
+import org.apache.hadoop.yarn.api.ApplicationConstants;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for the {@link YarnEntrypointUtils}.
+ */
+public class YarnEntrypointUtilsTest extends TestLogger {
+
+	private static final Logger LOG = LoggerFactory.getLogger(YarnEntrypointUtilsTest.class);
+
+	@ClassRule
+	public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+
+	/**
+	 * Tests that the REST ports are correctly set when loading a {@link Configuration}
+	 * with unspecified REST options.
+	 */
+	@Test
+	public void testRestPortOptionsUnspecified() throws IOException {
+		final File workingDirectory = TEMPORARY_FOLDER.newFolder();
+		final Configuration initialConfiguration = new Configuration();
+
+		final Configuration configuration = loadConfiguration(workingDirectory, initialConfiguration);
+
+		// having not specified the ports should set the rest bind port to 0
+		assertThat(configuration.getString(RestOptions.BIND_PORT), is(equalTo("0")));
+	}
+
+	/**
+	 * Tests that the binding REST port is set to the REST port if set.
+	 */
+	@Test
+	public void testRestPortSpecified() throws IOException {
+		final File workingDirectory = TEMPORARY_FOLDER.newFolder();
+		final Configuration initialConfiguration = new Configuration();
+		final int port = 1337;
+		initialConfiguration.setInteger(RestOptions.PORT, port);
+
+		final Configuration configuration = loadConfiguration(workingDirectory, initialConfiguration);
+
+		// if the bind port is not specified it should fall back to the rest port
+		assertThat(configuration.getString(RestOptions.BIND_PORT), is(equalTo(String.valueOf(port))));
+	}
+
+	/**
+	 * Tests that the binding REST port has precedence over the REST port if both are set.
+	 */
+	@Test
+	public void testRestPortAndBindingPortSpecified() throws IOException {
+		final File workingDirectory = TEMPORARY_FOLDER.newFolder();
+		final Configuration initialConfiguration = new Configuration();
+		final int port = 1337;
+		final String bindingPortRange = "1337-7331";
+		initialConfiguration.setInteger(RestOptions.PORT, port);
+		initialConfiguration.setString(RestOptions.BIND_PORT, bindingPortRange);
+
+		final Configuration configuration = loadConfiguration(workingDirectory, initialConfiguration);
+
+		// bind port should have precedence over the rest port
+		assertThat(configuration.getString(RestOptions.BIND_PORT), is(equalTo(bindingPortRange)));
+	}
+
+	@Nonnull
+	private Configuration loadConfiguration(File workingDirectory, Configuration initialConfiguration) throws IOException {
```

Comment:
* Could make `static`
* Could move `TEMPORARY_FOLDER.newFolder();` here and remove from tests as it's never used within the test

Reply:
Good points. Will change it.